parser/parser/cc/lexer.rl

```diff
@@ -2272,7 +2272,7 @@ void lexer::set_state_expr_value() {
           | '@@' %{ tm = p - 2; }
           ) [0-9]*
       => {
-        if (version == ruby_version::RUBY_27) {
+        if (version >= ruby_version::RUBY_27) {
```

Comment:
This was a small mismatch with the `whitequark/parser` that results in a bug when upgrading the lexer ruby version to 3.1.
https://github.com/whitequark/parser/blob/ec523c0f8e9eaedcfb892b8e64394efbce9777d8/lib/parser/lexer.rl#L1899

Included with this PR because I wanted to keep my other feature PR simpler but I can remove it since it's not that relevant.